app/src/main/java/de/westnordost/streetcomplete/data/meta/MetadataModule.kt

```diff
@@ -1,23 +1,16 @@
 package de.westnordost.streetcomplete.data.meta
 
 import android.content.res.AssetManager
-import dagger.Module
-import dagger.Provides
 import de.westnordost.countryboundaries.CountryBoundaries
 import de.westnordost.osmfeatures.AndroidFeatureDictionary
-import de.westnordost.osmfeatures.FeatureDictionary
+import org.koin.core.qualifier.named
+import org.koin.dsl.module
 import java.util.concurrent.FutureTask
-import javax.inject.Singleton
 
-@Module
-object MetadataModule {
+val metadataModule = module {
+    factory { AbbreviationsByLocale(get()) }
 
-    @Provides @Singleton fun countryInfos(assetManager: AssetManager, countryBoundaries: FutureTask<CountryBoundaries>): CountryInfos =
-        CountryInfos(assetManager, countryBoundaries)
-
-    @Provides @Singleton fun countryBoundariesFuture(assetManager: AssetManager): FutureTask<CountryBoundaries> =
-        FutureTask { CountryBoundaries.load(assetManager.open("boundaries.ser")) }
-
-    @Provides @Singleton fun featureDictionaryFuture(assetManager: AssetManager): FutureTask<FeatureDictionary> =
-        FutureTask { AndroidFeatureDictionary.create(assetManager, "osmfeatures/default", "osmfeatures/brands") }
+    single { CountryInfos(get(), get<FutureTask<CountryBoundaries>>(named("CountryBoundariesFuture"))) }
+    single(named("CountryBoundariesFuture")) { FutureTask { CountryBoundaries.load(get<AssetManager>().open("boundaries.ser")) } }
+    single(named("FeatureDictionaryFuture")) { FutureTask { AndroidFeatureDictionary.create(get(), "osmfeatures/default", "osmfeatures/brands") } }
```

Comment:
Why is a name necessary? Can Koin not handle generics? (Would it be different if it was not a `FutureTask` but a `Deferred`?)

Reply:
>Can Koin not handle generics

Correct, Koin saw both as FutureTask.  See also [Dealing with generics](https://insert-koin.io/docs/reference/koin-core/definitions/#dealing-with-generics)